app/models/agents/freme_ner_agent.rb

```diff
@@ -55,27 +58,29 @@ def default_options
     form_configurable :mode, type: :array, values: ['all', 'spot', 'spot,classify', 'spot,link', 'spot,link,classify', 'link']
     form_configurable :domain
     form_configurable :types
+    form_configurable :numLinks
 
     def validate_options
       errors.add(:base, "body needs to be present") if options['body'].blank?
       errors.add(:base, "base_url needs to be present") if options['base_url'].blank?
       errors.add(:base, "base_url needs to end with a trailing '/'") unless options['base_url'].end_with?('/')
       errors.add(:base, "dataset needs to be present") if options['dataset'].blank?
+      errors.add(:base, "number needs to be greater than 0 and less than or equal to 5") unless options['numLinks'].blank? || options['numLinks'] =~ /[0-5]/
```

Comment:
Interesting way of validating the number, but it that would allow `11` as well as `0`. I would suggest either `/\A[1-5]\z/` or `unless options['numLinks'].blank? || (1..5) === options['numLinks'].to_i`